src/models/model-types.ts

```diff
@@ -3,8 +3,18 @@ type TestDescriptorType = {
   __testIsInvalidSample?: boolean;
 };
 
+type ToString<T> = T extends Date
+  ? string
+  : T extends object
+  ? RawModel<T>
+  : string;
+
 export type TestDependencyDataType<T = unknown> = Array<TestDescriptorType & T>;
 
+export type RawModel<T> = {
```

Comment:
Can you explain what RawModel is?

Reply:
RawModel recursively types any property of a model into string properties, which can be useful for marking an unprocessed raw data model from the API that has not been normalized, I just couldn't figure a better generic name for it

```ts
interface Model {
  isSet: boolean;
  name: string;
  count: number;
  object: {
    comment: string;
    date: Date;
  };
}

// marking an object with RawModel results in all properties being typed into string
// i.e variable.count, variable.object.date, etc. are now typed as string
let variable: RawModel<Model>;
```

Reply:
And why do we need to do this?